.github/actions/label/build.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * Javascript module for the label action.
+ */
+
+const [owner, repo] = ["gear-tech", "gear"];
+const { LABEL, REF, HEAD_SHA } = process.env;
+const linux = LABEL === "A0-pleasereview";
+const checks = linux ? ["linux", "win-cross"] : ["x86"];
+const workflow_id = linux
+  ? ".github/workflows/build.yml"
+  : ".github/workflows/build-macos.yml";
+
+/**
+ *  Sleep for ms milliseconds.
+ **/
+const sleep = (ms) => new Promise((r) => setTimeout(r, ms));
+
+/**
+ *  If skipping this action.
+ **/
+const skip = async ({ github }) => {
+  const {
+    data: { check_runs },
+  } = await github.rest.checks.listForRef({
+    owner,
+    repo,
+    ref: REF,
+  });
+
+  const runs = linux
+    ? check_runs.filter((run) => run.name === "build" || run.name === "build / linux")
+    : check_runs.filter((run) => run.name === "build / macox-x86");
+
+  if (runs.length === 0) return false;
+  for (run of runs) {
+    if (run.conclusion !== "skipped") return true;
+  }
+
+  return false;
+};
+
+/**
+ *  Create build checks.
+ *
+ *  TODO:
+ *    * Queue the new created checks to check suite PR (#3087).
+ *    * Support re-runing the checks. (#3088)
+ **/
+const createChecks = async ({ core, github }) => {
+  let status = {};
+  for (check of checks) {
+    const { data: res } = await github.rest.checks.create({
+      owner,
+      repo,
+      name: `build / ${check}`,
+      head_sha: HEAD_SHA,
+    });
+
+    core.info(`Created check ${check}`);
+    status[check] = res;
+  }
+
+  return status;
+};
+
+/**
+ *  Dispatch the target workflow.
+ */
+const dispatchWorkflow = async ({ core, github }) => {
+  await github.rest.actions.createWorkflowDispatch({
+    owner,
+    repo,
+    workflow_id,
+    ref: REF,
+  });
+
+  await sleep(10000);
```

Comment:
Why sleep? And how this value had been chosen?

Reply:
There will be short delays inside the API service of gtihub after dispatching the workflow, 10s is insanely safe, in the next step `listWorkflowRuns` requires workflow info from the API service of github